### PR TITLE
[#158073462] add terms of service to Services and accepted ToS in user's Profile

### PR DESCRIPTION
**Pull request by gunzip**
in user profile we'll have

```
terms_of_service_accepted: {
   "someServiceId1": <numeric version>,
   "someServiceId2": <numeric version>,
   ...
}
```

### CODE REVIEW

File: api/definitions.yaml [hunk 1]
```diff
@@ -399,13 +379,54 @@ ServicePublic:
...
   required:
     - service_id
     - service_name
     - organization_name
     - department_name
+    - terms_of_service
+    - terms_of_service_version
+Service:
+  description: A Service tied to an user's subscription.
+  allOf:
```
Comment (gunzip) [inline api/definitions.yaml]: now it's possibile to use allOf

File: lib/controllers/__tests__/messages.test.ts [hunk 2]
```diff
@@ -93,9 +93,12 @@ const someUserAttributes: IAzureUserAttributes = {
     authorizedCIDRs: toAuthorizedCIDRs([]),
     authorizedRecipients: new Set([]),
     departmentName: "IT" as NonEmptyString,
+    fiscalCode: aFiscalCode,
```
Comment (gunzip) [inline lib/controllers/__tests__/messages.test.ts]: forked from https://github.com/teamdigitale/digital-citizenship-functions/pull/263

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Remember to fix the PR title by removing WIP wording when ready
  </td>
    </tr>
  </tbody>
</table>



## Affected stories

  * 🌟 [#158073462](https://www.pivotaltracker.com/story/show/158073462): Come CIT voglio poter memorizzare la versione dei TOS che ho sottoscritto, in modo da non doverli sottoscrivere più volte

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by gunzip** [comment]
now we'll have this kind of structure for a user profile:
 
```
{
	"blocked_inbox_or_channels": {
		"5a25abf4fcc89605c082f042c49a": [
			"WEBHOOK"
		]
	},
	"terms_of_service_accepted": {
		"5a25abf4fcc89605c082f042c49a": 1
	},
    "is_inbox_enabled": true,
    "is_webhook_enabled": true,
    "preferred_languages": [
        "it_IT"
    ],
    "version": 13
}
```

it looks like at the end we'll have, for each service, some kind of metadata in the user profile. now I'm wondering if it would be better to have a structure like the following (to have a one-to-many relationship profile -> services and to not repeat the serviceId everywhere):

```
{
   services: 
   {
		"5a25abf4fcc89605c082f042c49a": [
	               "blocked_inbox_or_channels": [ "WEBHOOK" ]
	                "terms_of_service_accepted": 1
	        },
    },
    "is_inbox_enabled": true,
    "is_webhook_enabled": true,
    "preferred_languages": [
        "it_IT"
    ],
    "version": 13
}
```

**Comment by gunzip** [comment]
now we'll have this kind of structure for a user profile:
 
```
{
  "blocked_inbox_or_channels": {
	"xxxxserviceid": [ "WEBHOOK" ]
   },
   "terms_of_service_accepted": {
	"xxxxserviceid": 1
    },
    "is_inbox_enabled": true,
    "is_webhook_enabled": true,
    "preferred_languages": [
        "it_IT"
    ],
    "version": 13
}
```

it looks like at the end we'll have, for each service, some kind of metadata in the user profile. now I'm wondering if it would be better to have a structure like the following (to have a one-to-many relationship profile -> services and to not repeat the serviceId everywhere):

```
{
   services: 
   {
		"xxxserviceid": [
	               "blocked_inbox_or_channels": [ "WEBHOOK" ]
	                "terms_of_service_accepted": 1
	        },
    },
    "is_inbox_enabled": true,
    "is_webhook_enabled": true,
    "preferred_languages": [
        "it_IT"
    ],
    "version": 13
}
```

@cloudify

**Comment by cloudify** [comment]
I would stay on the denormalized structure:

```json
{
  "blocked_inbox_or_channels": {
	"xxxxserviceid": [ "WEBHOOK" ]
   },
   "terms_of_service_accepted": {
	"xxxxserviceid": 1
    },
    "is_inbox_enabled": true,
    "is_webhook_enabled": true,
    "preferred_languages": [
        "it_IT"
    ],
    "version": 13
}
```

just `terms_of_service_accepted` -> `accepted_service_tos_version`

**Comment by cloudify** [comment]
Also, we should NOT keep the content of the tos in the services record, let's leave the app have the responsibility to gather the TOS from some source (that we don't know yet) - handling the content of the service TOS shouldn't be a responsibility of this API.

**Comment by gunzip** [comment]
> Also, we should NOT keep the content of the tos in the services record, let's leave to the app (and its backend) the responsibility to gather the TOS from some source (that we don't know yet) - handling the 
content of the service TOS shouldn't be a responsibility of this API.

so we should store a link (url) to tos or nothing at all ?

**Comment by cloudify** [comment]
nothing at all for now, we'll probably handle this by publishing something, statically, somewhere (e.g. in a github repo)

**Comment by cloudify** [comment]
this service will just be about keeping the state of each user

**Comment by gunzip** [comment]
> this service will just be about keeping the state of each user

it's not clear if we want to keep the latest version of ToS into the service metadata (if we don't, how do we match the latest version the uses has accepted with the latest published for the service ?)

**Comment by cloudify** [comment]
there will be a file, published somewhere with this format (`service_id` -> `version`):

```js
{
  "service_id_1": 1
}
```

for each service/version tuple there will be a corresponding markdown file, also published somewhere (e.g. `service_1/1.md`).

incidentally, all these files can be published on github and updated by each service through PRs

**Comment by gunzip** [comment]
right. then it's ready for review

**Pull request merged**